(almost) Complete rewrite of debugOverlay

LeviTaule opened the pull request:
I've made it much more generic.
- Adding and removing watched variables has been made much much easier (just using the API, there's no need to make any computations about the position or anything else)
- The exact way of rendering can be changed for all watched variables on one place, local method `renderVariable` (switching sides, changing colors, etc)
- This version is expected to be a bit more inefficient in terms of both memory and time, as it doesn't group variables from the same source.  However, it should not cause any performance issues (benchmarks needed).

LeviTaule committed:
Added assets to gitignore
So one can commit with no barriers.
The file may be removed in the future.

LeviTaule committed:
Fixed grammar for local variables in debugoverlay

LeviTaule committed:
Debug Overlay is now much more generic.
Adding and removing variables has been made much much easier (just using the API, there's no need to make any computations about the position or anything else)
The exact way of rendering can be changed for all watched variables on on place, local method renderVariable (switching sides, changing colores, etc)
This version is expected to be a bit more inefficent in terms of both memory and time, as it doesn't group variables from the same source.  However, it should not cause any performance issues (benchmarks needed).

zorggn commented:
From a first look, I can say that the code does indeed get more logical this way. Two very minor issues are that lagPercentage line is now misaligned with all the others, (though that one's moot because of the final commit), the other is that we could fit in the dynamic memory conversion routine into the custom function for texture memory. Bonus points for making me go wtf at my own english grammar usage. :3

LeviTaule commented:
OK. I actually want to fix some more things there (very minor) so I'll incorporate the points mentioned by you there tomorrow. Then we can finally move on :-) (Although I've noticed  that you have already begun doing that :-) )

LeviTaule committed:
TPS and FPS values are now displayed correctly
It was a minor fix of a copy&paste error

LeviTaule committed:
The dynamic conversion of the memory size is used from now on
I made the string slightly shorter. It's still easy to change it back, though. Also, it's not memoryprefix (B/kB/MB...), but rather memorypostfix (at least when using it like this, although, technically it doesn't matter, since it's multiplication, which is commutative anyway)
